Sources/CouchDB/CouchDBUtils.swift

```diff
@@ -77,16 +41,27 @@ class CouchDBUtils {
         return requestOptions
     }
 
-    class func getBodyAsJson (_ response: ClientResponse) -> JSON? {
+    class func getBodyAsCodable<O: Decodable> (_ response: ClientResponse) throws -> O {
         do {
             var body = Data()
             try response.readAllData(into: &body)
-            let json = JSON(data: body)
-            return json
+            let codable = try JSONDecoder().decode(O.self, from: body)
+            return codable
         } catch {
-            //Log this exception
+            throw error
+        }
+    }
+    
+    class func getBodyAsError(_ response: ClientResponse) -> CouchDBError {
+        do {
+            var body = Data()
+            try response.readAllData(into: &body)
+            var error = try JSONDecoder().decode(CouchDBError.self, from: body)
```

Comment:
A comment here explaining that error will be of type CouchDBError would be useful. I had to go look up JSONDecoder to work this out.

Reply:
Actually, ignore this!